lib/signs_ui/alerts/state.ex

```diff
@@ -4,87 +4,147 @@ defmodule SignsUi.Alerts.State do
   them with signs to auto-expire.
   """
 
-  use GenStage
+  use GenServer
 
   require Logger
-  alias ServerSentEventStage.Event
   alias SignsUi.Alerts.Alert
   alias SignsUi.Alerts.Display
-  alias SignsUi.Alerts.Events
 
   @type t :: %{Alert.id() => Alert.multi_route()}
 
   @spec start_link(keyword) :: :ignore | {:error, any} | {:ok, pid}
   def start_link(opts \\ []) do
-    {start_opts, init_opts} = Keyword.split(opts, [:name])
-    GenStage.start_link(__MODULE__, init_opts, start_opts)
+    GenServer.start_link(__MODULE__, [], opts)
   end
 
-  @spec active_alert_ids(GenStage.stage()) :: MapSet.t(Alert.id())
+  @spec active_alert_ids(SignsUi.Alerts.State) :: MapSet.t(Alert.id())
   def active_alert_ids(pid \\ __MODULE__) do
-    GenStage.call(pid, :active_alert_ids)
+    GenServer.call(pid, :active_alert_ids)
   end
 
-  @spec all(GenStage.stage()) :: Display.t()
+  @spec all(SignsUi.Alerts.State) :: Display.t()
   def all(pid \\ __MODULE__) do
-    GenStage.call(pid, :all)
+    GenServer.call(pid, :all)
   end
 
-  @impl GenStage
-  def init(opts) do
-    {:consumer, %{}, opts}
+  @impl GenServer
+  def init(_) do
+    Process.send(self(), :update, [])
+    {:ok, %{}}
   end
 
-  @impl GenStage
+  @impl GenServer
   def handle_call(:active_alert_ids, _from, state) do
     alert_ids = state |> Map.keys() |> MapSet.new()
 
-    {:reply, alert_ids, [], state}
+    {:reply, alert_ids, state}
   end
 
-  @impl GenStage
+  @impl GenServer
   def handle_call(:all, _from, state) do
-    {:reply, Display.format_state(state), [], state}
+    {:reply, Display.format_state(state), state}
   end
 
-  @impl GenStage
-  @spec handle_events([Event.t()], GenStage.from(), t()) :: {:noreply, [], t()}
-  def handle_events(events, _from, state) do
-    # Works in two primary phases. First, we generate fresh t using the
-    # provided events:
-    new_state = update_state(events, state)
-    Logger.info(["alert_state_updated ", inspect(new_state)])
-
-    # Next, we convert our internal model to the specified format:
-    display_state = Display.format_state(new_state)
-    SignsUiWeb.Endpoint.broadcast!("alerts:all", "new_alert_state", display_state)
-    {:noreply, [], new_state}
+  @impl GenServer
+  def handle_info(:update, state) do
+    state =
+      case fetch_alerts() do
+        {:ok, response} ->
+          new_state = Enum.into(parse(response), %{}, &{&1.id, &1})
+
+          SignsUiWeb.Endpoint.broadcast!(
+            "alerts:all",
+            "new_alert_state",
+            Display.format_state(new_state)
+          )
+
+          Logger.info(["alert_state_updated ", inspect(new_state)])
```

Comment:
note: This now logs every 5 seconds, regardless of whether there was a change. If we have any reports tied to this, we may need to update them.